spec.md

```diff
@@ -937,140 +783,208 @@ Property types are legal in the following constructs:
-Type aliases enables their reuse across multiple declarations.
+Type aliases allow us to give names to types.
+This is useful for bundle types, especially when they have many fields.
+It is also useful for hinting at what the value represents.
+
+Examples:
 
 ```firrtl
 type WordType = UInt<32>
 type ValidType = UInt<1>
-type Data = {w: WordType, valid: ValidType, flip ready: UInt<1>}
+type Data = { w : WordType, valid : ValidType, flip ready : UInt<1> }
 type AnotherWordType = UInt<32>
 
 module TypeAliasMod:
-  input in: Data
-  output out: Data
-  wire w: AnotherWordType
+  input in : Data
+  output out : Data
+  wire w : AnotherWordType
   connect w, in.w
-  ...
+  ; ...
 ```
 
-The `type` declaration is globally defined and all named types exist in the same namespace and thus must all have a unique name.
+Type aliases have structural identity.
+In other words, when we compare two types, we expand all type aliases recursively until we are left with type expressions that have no aliases.
+For instance, in the above example, we can connect `in.w`{.firrtl} to `w`{.firrtl} since their types, `WordType` and `AnotherWordType` respectively, both expand to `UInt<32>`{.firrtl}.
+
+The `type`{.firrtl} declaration is globally defined and all named types exist in the same namespace and thus must all have a unique name.
 Type aliases do not share the same namespace as modules; hence it is allowed for type aliases to conflict with module names.
-Note that when we compare two types, the equivalence is determined solely by their structures.
-For instance types of `w`{.firrtl} and `in.w`{.firrtl} are equivalent in the example above even though they are different type alias.
 
-## Type Equivalence
+## Type Inference
 
-The type equivalence relation is used to determine whether a connection between two components is legal.
-See [@sec:connects] for further details about connect statements.
+FIRRTL has support for limited type inference.
+This comes in two flavors:
+Width inference and reset inference.
 
-An unsigned integer type is always equivalent to another unsigned integer type regardless of bit width, and is not equivalent to any other type.
-Similarly, a signed integer type is always equivalent to another signed integer type regardless of bit width, and is not equivalent to any other type.
+### Width Inference
 
-Clock types are equivalent to clock types, and are not equivalent to any other type.
+Normally, the three integer ground types are written with explicit bit widths.
+The bit widths are given in angel brackets, such as the `8`{.firrtl} in `UInt<8>`{.firrtl}.
+This is called the **uninferred** variant of the type.
 
-An uninferred `Reset`{.firrtl} can be connected to another `Reset`{.firrtl}, `UInt`{.firrtl} of unknown width, `UInt<1>`{.firrtl}, or `AsyncReset`{.firrtl}.
-It cannot be connected to both a `UInt`{.firrtl} and an `AsyncReset`{.firrtl}.
+FIRRTL also supports an **inferred** variant of these types.
+They are written as follows:
```

Comment:
The copy here introduces `uninferred` for the first time. However, `inferred` was introduced in the discussion of types. Does it make sense to introduce `uninferred` earlier as well?